src/main/java/games/descent2e/DescentForwardModel.java

```diff
@@ -323,35 +323,132 @@ protected List<AbstractAction> _computeAvailableActions(AbstractGameState gameSt
 
```

Comment:
Some potential points on things that exist that could simplify some of this logic 
- Utils.getNeighbourhood(x,y,width,height,8-way connection) is a function that returns the list of Vector2D coordinates for adjacent squares (although checking the board node neighbours makes more sense here, just noting this in case it becomes useful at some point)
- Also, all board nodes on the grid have a property that stores the component ID of the figure at that location (-1 if empty I believe, or property is null or something, worth checking FM setup) - at least in the current setup. So can get e.g. a Figure object that occupies a board node by (Figure)gs.getComponentById(((PropertyInt)boardNode.getProperty(playersHash)).value). I think there are some checks for occupancy in your code, this might be an easier way to do that (can then potentially check if figure instanceof Hero/Monster to see if friendly). This might change in the future, but I think nodes should store figure occupancy anyway so some form of this should exist.

Reply:
Changed the logic of the code to use gs.getComponentById(((PropertyInt)boardNode.getProperty(playersHash)).value instead